src/http/request_handler.rs

```diff
@@ -105,11 +105,31 @@ fn create_allowed_headers(headers: &HeaderMap, allowed: &BTreeSet<String>) -> He
     new_headers
 }
 
+pub async fn preflight(app_ctx: Arc<AppContext>, req: Request<Body>) -> Result<Response<Body>> {
+    let _whole_body = hyper::body::aggregate(req).await?;
+    let mut response = Response::builder().status(StatusCode::OK);
+
+    for (key, val) in &app_ctx.blueprint.server.response_headers {
+        if key
+            .as_str()
+            .to_lowercase()
+            .starts_with("access-control-allow-")
+        {
+            response = response.header(key, val);
+        }
+    }
+
+    let response = response.body(Body::default())?;
+
+    Ok(response)
+}
+
 pub async fn handle_request<T: DeserializeOwned + GraphQLRequestLike>(
     req: Request<Body>,
     app_ctx: Arc<AppContext>,
 ) -> Result<Response<Body>> {
     match *req.method() {
+        hyper::Method::OPTIONS => preflight(app_ctx.clone(), req).await,
```

Comment:
probably, not every OPTIONS request should be considered as CORS request. The client in case of CORS provides some additional headers like https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS